src/pages/__test__/cart.test.js

```diff
@@ -0,0 +1,855 @@
+import '@testing-library/jest-dom/extend-expect';
+import React from 'react';
+import * as router from 'react-router';
+import { waitFor, render, screen, fireEvent } from '@testing-library/react';
+import AppContext from '../../utils/appContext';
+import * as AppContextFile from '../../utils/appContext';
+import ShoppingCart from '../../components/ShoppingCart/shoppingCart';
+
+describe('Product List Page tests: ', () => {
+    const navigate = jest.fn();
+    let confirmSpy;
+    beforeAll(() => {
+        confirmSpy = jest.spyOn(window, 'confirm');
+        confirmSpy.mockImplementation(jest.fn(() => true));
+    });
+    beforeEach(() => {
+        jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test('Validate that an empty state is displayed when there are no items in the cart.', async () => {
+        jest.spyOn(AppContextFile, 'useAppContext').mockReturnValue({
+            dispatch: jest.fn(),
+            state: {
+                products: []
+            }
+        });
+        const mockDispatch = jest.fn();
+        render(
+            <AppContext.Provider value={mockDispatch}>
+                <ShoppingCart />
```

Comment:
The recommendation of using `msw` is that you need to render the `App` component and then do the navigation to the `Cart` page, the intention of this is to try to simulate a similar behavior as the users do in the browsers.